selene_sdk/samplers/random_positions_sampler.py

```diff
@@ -282,6 +299,66 @@ def _retrieve(self, chrom, position):
                 self.save_dataset_to_file(self.mode)
         return (retrieved_seq, retrieved_targets)
 
+# Make a dictionary of chromosomes which maps to how many elements from chrom
+# we sample. Determined probabilistically and weighted by proportion.
+    def _samples_per_chrom(self):
+        proportions = np.subtract(self.chrom_ends, self.chrom_starts)
+        proportions[:] = [chr_len / self.genome_length for chr_len in proportions]
+        samples_per_chrom = defaultdict(int)
+
+        for i in range(self.sample_size):
+            chr = np.random.choice(self.chroms, p=proportions)
+            samples_per_chrom[chr] += 1
+
+        return samples_per_chrom
+
+# Updates tot_sample to hold the indices we will sample from
+# of a hypothetical flattened array of each chromosome laid side by side.
+# These indices can then be converted into (chrom, position) pairs to sample.
+    def _generate_sample(self):
+        tot_len = 0
+        for chrom, len_chrom in self.reference_sequence.get_chr_lens():
+            self.chroms = np.append(self.chroms, chrom)
+            self.chrom_starts = np.append(self.chrom_starts, tot_len)
+            self.chrom_ends = np.append(self.chrom_ends, tot_len + len_chrom - 1)
+            tot_len += len_chrom
+
+        self.genome_length = tot_len
+
+        samples_per_chrom = self._samples_per_chrom()
+        for chrom, len_chrom in self.reference_sequence.get_chr_lens():
+            chrom_range = np.arange(tot_len, tot_len + len_chrom - 1)
```

Comment:
this looks like the section where you are adding samples to `tot_len`, I would recommend printing out what exactly is the chrom range you are specifying and what is being returned by np.random.choice